File: app/src/main/java/com/funckyhacker/fileexplorer/view/search/SearchViewModel.java
package com.funckyhacker.fileexplorer.view.search;

import com.funckyhacker.fileexplorer.view.adapter.MainLinearAdapter;
import java.io.File;
import java.util.List;
import rx.Observable;

public abstract class SearchViewModel {

  public abstract Observable<List<File>> search(String query);

  public abstract void init();

  public abstract MainLinearAdapter getLinearAdapter();

  public abstract void setData(List<File> files);
}

File: app/src/main/java/com/funckyhacker/fileexplorer/view/search/SearchViewModel.kt
package com.funckyhacker.fileexplorer.view.search

import com.funckyhacker.fileexplorer.view.adapter.MainLinearAdapter
import rx.Observable
import java.io.File

abstract class SearchViewModel {

    abstract val linearAdapter: MainLinearAdapter

    abstract fun search(query: String): Observable<List<File>>

    abstract fun init()

    abstract fun setData(files: List<File>)
}
